File: rb/problem15.rb
result = 0
def calcRoute(width, height)
    if width == 0 or height == 0 then
        p width, height
        result += 1
    elsif width < 2 or height < 2 then
        if width == 1 then
            result += height + 1
        elsif height == 1 then
            result += width + 1
        end
    else
        height.times do |n|
            calcRoute(width-1, n + 1)
        end
    end
end

p calcRoute(0, 0)

File: py/problem15.py
sum = 0
def calcRoute(width_, height_) :
    global sum
    if width_ >= 2 :
        for n in range(height_ + 1) :
            calcRoute(width_ - 1, n)
    elif width_ == 1 :
        sum += height_ + 1

    return sum

print(calcRoute(20, 20))
